src/me/mrCookieSlime/Slimefun/Setup/SlimefunManager.java

```diff
@@ -123,20 +131,33 @@ else if (!item.getItemMeta().hasDisplayName() && !sfitem.getItemMeta().hasDispla
 	public static boolean isItemSimiliar(ItemStack item, ItemStack sfitem, boolean lore, DataType data) {
 		return isItemSimiliar(item, sfitem, lore);
 	}
+
+	public static boolean containsSimilarItem(Inventory inventory, ItemStack itemStack, boolean checkLore) {
+		if (inventory == null || itemStack == null) return false;
+
+		for (ItemStack is : inventory.getStorageContents()) {
+			if (is == null || is.getType() == Material.AIR) continue;
+			if (isItemSimiliar(is, itemStack, checkLore))
+				return true;
+		}
+
+		return false;
+	}
 	
 	private static boolean equalsLore(List<String> lore, List<String> lore2) {
-		StringBuilder string1 = new StringBuilder();
-		StringBuilder string2 = new StringBuilder();
+		Iterator<String> original = lore.iterator();
+		Iterator<String> comparator = lore2.iterator();
 
-		String colors = ChatColor.YELLOW.toString() + ChatColor.YELLOW.toString() + ChatColor.GRAY.toString();
-		for (String string: lore) {
-			if (!string.equals(ChatColor.GRAY + "Soulbound") && !string.startsWith(colors)) string1.append("-NEW LINE-").append(string);
-		}
-		
-		for (String string: lore2) {
-			if (!string.equals(ChatColor.GRAY + "Soulbound") && !string.startsWith(colors)) string2.append("-NEW LINE-").append(string);
+		while (original.hasNext() && comparator.hasNext()) {
```

Comment:
This condition isn't really a good design choice.
If one of the items has fewer lines of lore than the other, the method will return true prematurely.